src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+            dustEffect->accel.z = randomNumber - 0.2f;
+            dustEffect->pos.x += dustEffect->velocity.x;
+            dustEffect->pos.y += dustEffect->velocity.y;
+            dustEffect->pos.z += dustEffect->velocity.z;
+            dustEffect->velocity.x += dustEffect->accel.x;
+            dustEffect->velocity.y += dustEffect->accel.y;
+            dustEffect->velocity.z += randomNumber - 0.2f;
+            dustEffect->scale += dustEffect->scaleStep;
+        }
+    }
+}
+
+void EnGo2_DrawDust(EnGo2* this, GlobalContext* globalCtx) {
+    EnGoEffect* dustEffect = this->dustEffects;
+    s16 alpha;
+    s16 firstDone;
+    s16 index;
+    s16 i;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_go2_eff.c", 111);
+
+    firstDone = false;
+    func_80093D84(globalCtx->state.gfxCtx);
+    if (1) {}
+
+    for (i = 0; i < ARRAY_COUNT(this->dustEffects); i++, dustEffect++) {
+        if (dustEffect->type) {
+            if (!firstDone) {
+                POLY_XLU_DISP = Gfx_CallSetupDL(POLY_XLU_DISP, 0);
+                gSPDisplayList(POLY_XLU_DISP++, &D_0600FD40);
+                gDPSetEnvColor(POLY_XLU_DISP++, 100, 60, 20, 0);
+                firstDone = true;
+            }
+
+            alpha = dustEffect->timer * (255.0f / dustEffect->initialTimer);
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 170, 130, 90, alpha);
+            gDPPipeSync(POLY_XLU_DISP++);
+            Matrix_Translate(dustEffect->pos.x, dustEffect->pos.y, dustEffect->pos.z, MTXMODE_NEW);
+            func_800D1FD4(&globalCtx->mf_11DA0);
+            Matrix_Scale(dustEffect->scale, dustEffect->scale, 1.0f, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_go2_eff.c", 137),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+            index = dustEffect->timer * (8.0f / dustEffect->initialTimer);
+            gSPSegment(POLY_XLU_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(sDLists[index]));
+            gSPDisplayList(POLY_XLU_DISP++, &D_0600FD50);
+        }
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_go2_eff.c", 151);
+}
+
+s32 EnGo2_SpawnDust(EnGo2* this, u8 initialTimer, f32 scale, f32 scaleStep, s32 numDustEffects, f32 radius,
+                    f32 yAccel) {
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    s32 i;
+    s16 angle;
+
+    pos = sPos;
+    velocity = sVelocity;
+    accel = sAccel;
+    pos = this->actor.posRot.pos;
+    pos.y = this->actor.groundY;
+    angle = (Math_Rand_ZeroOne() - 0.5f) * 0x10000;
+
+    i = numDustEffects;
+    while (i >= 0) {
+        accel.y += Math_Rand_ZeroOne() * yAccel;
+        pos.x = (Math_Sins(angle) * radius) + this->actor.posRot.pos.x;
+        pos.z = (Math_Coss(angle) * radius) + this->actor.posRot.pos.z;
+        EnGo2_AddDust(this, &pos, &velocity, &accel, initialTimer, scale, scaleStep);
+        angle += (s16)(0x10000 / numDustEffects);
+        i--;
+    }
```

Comment:
I'm sure you've probably checked this but can this while loop be a for loop?
`for (i = numDustEffects; i >= 0; i--)`

Reply:
It sort of can, the for-loop was how it was originally. But for that form to match, everything had to be in a single line. There's a conversation about it somewhere in decomp-help but my solution was to follow the function EnFw_SpawnDust and move it to a while loop to get a match that didn't have the entire loop in a single line

Reply:
https://discordapp.com/channels/688807550715560050/688851337085190255/788195777607761981

Reply:
The while loop is perfectly readable so it's not really a problem, it's just weird that the `for` loop doesn't play nicely. I'd leave this open for a little bit longer in case anyone has any more suggestions but if not then you can resolve it